openmetadata-ui/src/main/resources/ui/src/components/Database/TableQueries/TableQueries.tsx

```diff
@@ -156,69 +205,274 @@ const TableQueries: FC<TableQueriesProp> = ({
       showErrorToast(error as AxiosError);
     }
   };
-  const fetchTableQuery = async (
-    params?: ListQueriesParams,
-    activePage?: number
-  ) => {
+
+  const handleSelectedQuery = (query: Query) => {
+    if (query.id !== selectedQuery?.id) {
+      setIsLoading((pre) => ({ ...pre, rightPanel: true }));
+      setSelectedQuery(query);
+      history.push({
+        search: Qs.stringify({
+          ...searchParams,
+          query: query.id,
+        }),
+      });
+    }
+  };
+
+  const fetchFilteredQueries = async (data?: {
```

Comment:
Recommend separating out business logic to util file to have cleaner component approch